src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,131 @@
+                }
+            case "event":
+                if (!sc.hasNextLine()) throw new DukeNoDescriptionException(w1);
+                else {
+                    String task = sc.nextLine().strip();
+                    String[] details = task.split(" /at ");
+                    if (details.length < 2) {
+                        throw new DukeNoDateException(w1);
+                    } else {
+                        isValidDateTime(details[1]);
+                        return new EventCommand(details[0], details[1]);
+                    }
+                }
+            case "deadline":
+                if (!sc.hasNextLine()) throw new DukeNoDescriptionException(w1);
+                else {
+                    String task = sc.nextLine().strip();
+                    String[] details = task.split(" /by ");
+                    if (details.length < 2) {
+                        throw new DukeNoDateException(w1);
+                    } else {
+                        isValidDateTime(details[1]);
+                        return new DeadlineCommand(details[0], details[1]);
+                    }
+                }
+            default:
+                throw new DukeUnknownCommandException();
+        }
+    }
+
+    public static Task parseFileLine(String e) throws DukeException {
+        String task_type = e.substring(0, 3);
+        String task_status = e.substring(3, 6);
+        String task_details = e.substring(7);
+        Task t;
+        switch (task_type) {
+            case "[T]":
+                t = new ToDo(task_details);
+                break;
+            case "[E]": {
+                String[] details = task_details.split(" /at ");
+                t = new Event(details[0], details[1]);
+                break;
+            }
+            case "[D]": {
+                String[] details = task_details.split(" /by ");
+                t = new Deadline(details[0], details[1]);
+                break;
+            }
+            default:
+                throw new DukeBadFileException();
+        }
+        if (task_status.equals("[X]")) {
+            t.setDone();
+        }
+        return t;
+    }
+
+    /*
+    IMPORTANT: parseFileLine incomplete, not all edge cases accounted for.
+     */
+
+    public static void isValidIndex(int index) throws DukeBadIndexException {
+        if (index == 0 || index < 0) {
+            throw new DukeBadIndexException(index);
+        }
+    }
+
+    public static void isValidDateTime(String dateTime) throws DukeBadDateException, DukeBadDateTimeException {
+        String[] p = dateTime.split(" ");
+        if (p.length < 2) {
+            throw new DukeBadDateTimeException();
+        } else {
+            String date = p[0];
+            String time = p[1];
+            String[] date_parsed = date.split("-");
+            if (date_parsed.length < 3) {
+                throw new DukeBadDateException();
+            }
+        }
+    }
+}
```

Comment:
I think according to the JavaDoc now for switch-case we don't have to indent inwards for "case". So it would have the same indentation as "switch" :)

Reply:
Yep! I changed it in my later commits haha